src/compiler/moduleNameResolver.ts

```diff
@@ -1899,6 +1920,8 @@ namespace ts {
                 return extension === Extension.Json;
             case Extensions.TypeScript:
                 return extension === Extension.Ts || extension === Extension.Tsx || extension === Extension.Dts;
+            case Extensions.TsOnly:
+                return extension === Extension.Ts || extension === Extension.Tsx;

```

Comment:
This is probably unrelated to this change, per sey, but this function probably needs to include the `.mts` and `.cts` flavor extensions in its logic. It was probably missed because it's only used in one place. :S

Reply:
Updated